internal/parse/parts.go

```diff
@@ -43,18 +43,25 @@ func (p *InputPart) ToSQL() string {
 // as well as the source table and column where it will be read from.
 type OutputPart struct {
 	Source []FullName
-	Target FullName
+	Target []FullName
 }
 
 func (p *OutputPart) String() string {
-	var colString string
+	var sourceString string
 	for _, col := range p.Source {
-		colString = colString + col.String() + " "
+		sourceString = sourceString + col.String() + " "
```

Comment:
The best way to implement the logic in this function is with a bytes.Buffer, otherwise a lot of unnecessary garbage is generated due to the allocation and release of all these immutable strings. Alternatively, why aren't we just using the `%v` verb of fmt.Sprintf for all these strings?  It looks very close, and it's zero effort on our side.

Reply:
Currently we don't use `fmt.Sprintf` so that we can make it more readable. For an expression &Person.* we would get "FullName{Prefix=Person, Name=*}" with Sprintf but instead we simply print "Person.*".

I'll have a look into changing to use bytes.Buffer, that looks neat

Reply:
> For an expression &Person.* we would get "FullName{Prefix=Person, Name=}"

Not really. Try it out. I think it'll help you, even if you use bytes.Buffer, so you don't need to do everything by hand.

Reply:
Though I've now just realised the Sprintf will use your custom string function so it does actually work very nicely! I've changed the String for ParsedExpr to use the bytes.Buffer pattern though

Reply:
That wasn't actually the point.. have you tried to use the `%v` verb on a struct?  The output is not the one suggested above.